tools/nitrogfx/gfx.c

```diff
@@ -297,6 +359,98 @@ void WriteImage(char *path, int numTiles, int bitDepth, int metatileWidth, int m
 	free(buffer);
 }
 
+void WriteNtrImage(char *path, int numTiles, int bitDepth, int metatileWidth, int metatileHeight, struct Image *image, bool invertColors, bool clobberSize, bool byteOrder)
+{
+    FILE *fp = fopen(path, "wb");
+
+    if (fp == NULL)
+        FATAL_ERROR("Failed to open \"%s\" for writing.\n", path);
+
+    int tileSize = bitDepth * 8;
+
+    if (image->width % 8 != 0)
+        FATAL_ERROR("The width in pixels (%d) isn't a multiple of 8.\n", image->width);
+
+    if (image->height % 8 != 0)
+        FATAL_ERROR("The height in pixels (%d) isn't a multiple of 8.\n", image->height);
+
+    int tilesWidth = image->width / 8;
+    int tilesHeight = image->height / 8;
+
+    if (tilesWidth % metatileWidth != 0)
+        FATAL_ERROR("The width in tiles (%d) isn't a multiple of the specified metatile width (%d)", tilesWidth, metatileWidth);
+
+    if (tilesHeight % metatileHeight != 0)
+        FATAL_ERROR("The height in tiles (%d) isn't a multiple of the specified metatile height (%d)", tilesHeight, metatileHeight);
+
+    int maxNumTiles = tilesWidth * tilesHeight;
+
+    if (numTiles == 0)
+        numTiles = maxNumTiles;
+    else if (numTiles > maxNumTiles)
+        FATAL_ERROR("The specified number of tiles (%d) is greater than the maximum possible value (%d).\n", numTiles, maxNumTiles);
+
+    int bufferSize = numTiles * tileSize;
+    unsigned char *pixelBuffer = malloc(bufferSize);
+
+    if (pixelBuffer == NULL)
+        FATAL_ERROR("Failed to allocate memory for pixels.\n");
+
+    int metatilesWide = tilesWidth / metatileWidth;
+
+    switch (bitDepth) {
+        case 4:
+            ConvertToTiles4Bpp(image->pixels, pixelBuffer, numTiles, metatilesWide, metatileWidth, metatileHeight, invertColors);
+            break;
+        case 8:
+            ConvertToTiles8Bpp(image->pixels, pixelBuffer, numTiles, metatilesWide, metatileWidth, metatileHeight, invertColors);
+            break;
+    }
+
+    WriteGenericNtrHeader(fp, "RGCN", bufferSize + 0x20, byteOrder);
+
+    unsigned char charHeader[0x20] = { 0x52, 0x41, 0x48, 0x43, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                                       0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                                       0x00, 0x00, 0x18, 0x00, 0x00, 0x00 };
```

Comment:
Line breaks after 13 bytes instead of 16 bytes -- why??

Reply:
120 character soft limit